API/trade_trees/controllers/trade_trees_controller.py

```diff
@@ -1,37 +1,68 @@
 from flask import Blueprint, request
 import flask
+from API.authentication.auth import auth
+#from flask_httpauth import HTTPBasicAuth as auth
 from kink import inject
 from schema import SchemaError
 from API.trade_trees.controllers.validation.trade_tree_validator import TradeTreeValidator
 from API.trade_trees.services.trade_tree_service import TradeTreeService
 from API.trade_trees.dto.trade_tree_parser import TradeTreeParser
-
-
+from API.users.services.user_service import UserService
+#from werkzeug.security import check_password_hash
+
+
+# auth = HTTPBasicAuth()
+#
+# USER_DATA = {
+#     "username": "password"
+# }
+#
+#
+# @auth.verify_password
+# def verify_password(username, password):
+#     if username in USER_DATA and \
+#             USER_DATA.get(username) == password:
+#         return username
 # Design notes:
 # Layer purposed for handling communication between client and API.
+
+
 @inject
 class TradeTreesController():
-    def __init__(self, trade_tree_service: TradeTreeService):
-        self.service = trade_tree_service
+    def __init__(self, trade_tree_service: TradeTreeService, user_service: UserService):
+        self.tree_service = trade_tree_service
+        self.user_service = user_service
         self.blueprint = self.define_routes()
         self.parser = TradeTreeParser()
         self.validator = TradeTreeValidator()
 
+    # @auth.verify_password
+    # def verify_password(self, username, password):
+    #     user = self.user_service.get_user(username)
+    #
+    #     if user is not None:
+    #         if check_password_hash(user.password, password):
+    #             return user
+    #     return None
+
     def define_routes(self):
         blueprint = Blueprint('trade_tree', __name__)
 
         @blueprint.route('/api/trade_tree/initialize', methods=['GET'])
+        @auth.login_required
         def initialize_trade_tree_table():
-            self.service.initialize_trade_tree_table()
+            self.tree_service.initialize_trade_tree_table()
```

Comment:
We're not fetching the trees by querying the specific user id from claims of the user (e.g. anyone can get anyone's tree and that violates OWASP top 10)